runtime/j9vm/jvm.c

```diff
@@ -1817,6 +1818,33 @@ jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args) {
 #else /* CALL_BUNDLED_FUNCTIONS_DIRECTLY */
 			zipFuncs = (J9ZipFunctionTable*) J9_GetInterface(IF_ZIPSUP, &j9portLibrary, j9binBuffer);
 #endif /* CALL_BUNDLED_FUNCTIONS_DIRECTLY */
+#ifdef WIN32
+			{
+				BOOLEAN conversionSucceed = FALSE;
+				int32_t size = 0;
+			
+				PORT_ACCESS_FROM_PORT(&j9portLibrary);
+				/* specialArgs.executableJarPath is retrieved from JavaVMInitArgs.options[i].optionString
+				 * which was set by Java Launcher in system default code page encoding.
+				 */
+				size = j9str_convert(J9STR_CODE_WINDEFAULTACP, J9STR_CODE_MUTF8, specialArgs.executableJarPath, strlen(specialArgs.executableJarPath), NULL, 0);
+				if (size > 0) {
+					size += 1; /* leave room for null */
+					executableJarPath = j9mem_allocate_memory(size, OMRMEM_CATEGORY_VM);
+					if (NULL != executableJarPath) {
+						size = j9str_convert(J9STR_CODE_WINDEFAULTACP, J9STR_CODE_MUTF8, specialArgs.executableJarPath, strlen(specialArgs.executableJarPath), executableJarPath, size);
+						if (size > 0) {
+							conversionSucceed = TRUE;
+						}
+					}
+				}
+				if (!conversionSucceed) {
+					result = JNI_ERR;
+					goto exit;
+				}
+				specialArgs.executableJarPath = executableJarPath;
```

Comment:
This code overwrites the original value of `specialArgs.executableJarPath`.  Who is responsible for freeing that memory?

Reply:
Original `specialArgs.executableJarPath` is from https://github.com/eclipse/openj9/blob/5a3774c9e698f1e0661420efebc0d93dd6075d1e/runtime/j9vm/jvm.c#L1309
which points to the `optionString`
https://github.com/eclipse/openj9/blob/5a3774c9e698f1e0661420efebc0d93dd6075d1e/runtime/j9vm/jvm.c#L1292

The `optionString` is going to be freed https://github.com/eclipse/openj9/blob/5a3774c9e698f1e0661420efebc0d93dd6075d1e/runtime/util/vmargs.c#L1621-L1642

Reply:
Updated with a comment.